library/Rules/Uuid.php

```diff
@@ -34,41 +37,44 @@ final class Uuid extends Standard
 {
     public const TEMPLATE_VERSION = '__version__';
 
-    private const PATTERN_FORMAT = '/^[0-9a-f]{8}-[0-9a-f]{4}-%s[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
-
     public function __construct(
         private readonly ?int $version = null
     ) {
+        if (!class_exists(RamseyUuid::class)) {
+            throw new MissingComposerDependencyException('Uuid rule requires ramsey/uuid', '=');
+        }
         if ($version !== null && !$this->isSupportedVersion($version)) {
             throw new InvalidRuleConstructorException(
-                'Only versions 1, 3, 4, and 5 are supported: %d given',
-                (string) $version
+                'Only versions 1 to 8 are supported: %d given',
+                (string)$version
             );
         }
     }
 
     public function evaluate(mixed $input): Result
     {
-        $template = $this->version ? self::TEMPLATE_VERSION : self::TEMPLATE_STANDARD;
+        $template   = $this->version ? self::TEMPLATE_VERSION : self::TEMPLATE_STANDARD;
         $parameters = ['version' => $this->version];
-        if (!is_string($input)) {
+
+        if (!is_string($input) && !($input instanceof UuidInterface)) {
             return Result::failed($input, $this, $parameters, $template);
         }
 
-        return new Result(preg_match($this->getPattern(), $input) > 0, $input, $this, $parameters, $template);
-    }
+        try {
+            $uuid = RamseyUuid::fromString((string) $input);
+        } catch (Throwable) {
+            return Result::failed($input, $this, $parameters, $template);
+        }
 
-    private function isSupportedVersion(int $version): bool
-    {
-        return $version >= 1 && $version <= 5 && $version !== 2;
+        /** @phpstan-ignore-next-line */
+        $uuidVersion = $uuid->getFields()->getVersion();
```

Comment:
I'm a bit concerned about ignoring this part here, because that means that `RamseyUuid::fromString` might return something that's not what we expecting. Isn't there a programatic way to verify this?

Reply:
I think that solution you propose is much better than ignoring the problem! Awesome!